plugins/modules/site_workflow_manager.py

```diff
@@ -787,81 +1356,229 @@ def get_diff_merged(self, config):
+                    site_updated = True
+
+                if site_updated and response and isinstance(response, dict):
+                    taskid = response["response"]["taskId"]
+                    task_details = self.get_task_details(taskid)
+                    while True:
+                        if task_details.get("progress") == (
+                            "Group is updated successfully"
+                            if site_type != "floor"
+                            else "Service domain is updated successfully."
+                        ):
+                            self.result['changed'] = True
+                            self.result['response'] = task_details
+                            break
+                        elif task_details.get("bapiError"):
+                            self.module.fail_json(msg=task_details.get(
+                                "bapiError"), response=task_details)
+                            break
                 else:
-                    # Get the site id of the newly created site.
-                    (site_exists, current_site) = self.site_exists()
+                    self.update_not_neeeded_sites.append(site_name)
+                    self.log(
+                        "Site - {0} does not need any update".format(site_name), "INFO")
+                    return self
 
+        else:
+            self.log(site_updated)
+
+            if self.dnac_version <= self.version_2_3_5_3:
+                try:
+                    site_params = self.want.get("site_params")
+                    try:
+                        if site_params['site'].get('building'):
+                            building_details = {
+                                key: value for key, value in site_params['site']['building'].items() if value is not None}
+                            site_params['site']['building'] = building_details
+                    except Exception as e:
+                        site_type = site_params['type']
+                        name = site_params['site'][site_type]['name']
+                        self.log(
+                            "The site '{0}' is not categorized as a building; no need to filter 'None' values.".format(name), "INFO")
+
+                    response = self.dnac._exec(
+                        family="sites",
+                        function='create_site',
+                        op_modifies=True,
+                        params=site_params,
+                    )
+                    self.log("Received API response from 'create_site': {0}".format(
+                        str(response)), "DEBUG")
+                    site_created = True
+                except Exception as e:
+                    self.log("Unexpected error occurred: {0}".format(
+                        str(e)), "ERROR")
+                    error_message = (
+                        "The Catalyst Center user '{0}' does not have the necessary permissions to 'create or update' "
```

Comment:
What is necessary permission ?

Reply:
If an update is not needed, I added it for that purpose and again changed it to the proper message.